sig-etcd/charter.md

```diff
@@ -0,0 +1,69 @@
+# SIG etcd Charter
+
+This charter adheres to the conventions described in the [Kubernetes Charter README] and uses
+the Roles and Organization Management outlined in [sig-governance].
+
+[Kubernetes Charter README]: https://github.com/kubernetes/community/blob/master/committee-steering/governance/README.md
+[sig-governance]: https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance.md
+
+## Scope
+
+Owns the etcd project and how it is used by Kubernetes.
+
+### In scope
+
+#### Code, Binaries and Services
+
+- Development of [etcd] and other repositories under [etcd-io organization]
+- Maintenance of [etcd image] packaged with Kubernetes
+
+[etcd]: https://github.com/etcd-io/etcd
+[etcd-io organization]: https://github.com/etcd-io
+[etcd image]: https://github.com/kubernetes/kubernetes/tree/master/cluster/images/etcd
+
+#### Cross-cutting and Externally Facing Processes
+
+- Specifying, testing and improving [The Implicit Kubernetes-ETCD Contract]
+- Release process of etcd and other binaries belonging to [etcd-io organization]
+
+[The Implicit Kubernetes-ETCD Contract]: https://docs.google.com/document/d/1NUZDiJeiIH5vo_FMaTWf0JtrQKCx0kpEaIIuPoj9P6A/edit?usp=sharing
+
+### Out of scope
+
+- Structure of data stored in etcd is owned by SIG API Machinery
+
+## Roles and Organization Management
+
+This SIG follows adheres to the Roles and Organization Management outlined in [sig-governance]
+and opts-in to updates and modifications to [sig-governance].
+
+### Additional responsibilities of Tech Leads
+
+- Release of etcd and other binaries belonging to [etcd-io organization]
+
+### Deviations from [sig-governance]
+
+- SIG etcd does not participate in release planning meetings,
+  retrospectives and burn-down meetings as etcd release process is independent of Kubernetes.
+- SIG etcd communication utilizes pre-existing forums for communication:
+  - Email: [etcd-dev](https://groups.google.com/forum/?hl=en#!forum/etcd-dev).
+  - Slack: [#etcd](https://kubernetes.slack.com/messages/C3HD8ARJ5/details/) channel on Kubernetes.
+- SIG etcd contributing instructions ([CONTRIBUTING.md]) be defined in etcd project.
+- KEPs created by SIG etcd will not follow Kubernetes release cycle and freezes due.
+
+Tech Leads must also fulfill all of the responsibilities of the Chair role as outlined in [sig-governance].
+
+[CONTRIBUTING.md]: https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md
+
+### Deviations from [kubernetes-repositories]
+
+- SIG etcd repositories live in github.com/etcd-io
+- SIG etcd repositories should (but not must) adopt merge bot, Kubernetes PR commands/bot, OWNERS file.
```

Comment:
This will require our contributors to apply for Kubernetes Org membership.  That's probably a good thing overall, but will involve some bootstrapping.

Reply:
That's not true, I expect similar to Kubernetes Org vs Kubernetes-SIGs Org, the membership will be separated from Kubernetes.

Reply:
It's not, even kubernetes sigs have to apply for membership through org.

Reply:
yup - our orgs all use the same process of applying. Essentially you apply and if approved you can join any of the primary k8s orgs.

I mentioned out of band (https://github.com/kubernetes/community/pull/7372#issuecomment-1633233709) but to directly comment here on some of the hard reqs:

- etcd org is owned and managed by k8s OR repos are transferred to K8s org (NOTE: people can still be granted repo admin rights, its just management of the org)
- membership will be managed by our our automation and applying for org membership will use our process (current etcd org members wouldn't have to apply, we'd directly add)
- EasyCLA will be enabled on the etcd org or any repos transferred to K8s org
- github actions or the CI currently in play can still be used, but we'd still have to run our global stuff (e.g. cncf-cla, lgtm/approve etc.)

Reply:
Thanks for bringing this up, Josh. I have no problem with that.

Reply:
One comment on the owners files, I think those will be a "must" insofar as we can link to the people responsible for the management of the subproject.

Reply:
I am ok with that, do we just need a k8s style OWNERS file in the top level of each subproject under https://github.com/etcd-io?

cc @serathius @ahrtr @spzala @jmhbnz

Reply:
Sounds good to me too!

Reply:
+1 sounds good to me.

Reply:
Removed the "OWNERS file" from deviation, and added an issue in etcd repo to track the effort of adding top level OWNERS file in each subproject repo: https://github.com/etcd-io/etcd/issues/16367